packages/@sanity/form-builder/src/inputs/files/ImageInput/ImageInput.tsx

```diff
@@ -552,30 +558,41 @@ export default class ImageInput extends React.PureComponent<Props, ImageInputSta
 
     return (
       <WithReferencedAsset observeAsset={observeAsset} reference={asset}>
-        {(assetDocument) => (
-          <ImageActionsMenu
-            isMenuOpen={isMenuOpen}
-            onEdit={this.handleOpenDialog}
-            showEdit={showAdvancedEditButton}
-            onMenuOpen={(isOpen) => this.setState({isMenuOpen: isOpen})}
-          >
-            <ActionsMenu
-              onUpload={this.handleSelectFiles}
-              browse={browseMenuItem}
-              onReset={this.handleRemoveButtonClick}
-              downloadUrl={imageUrlBuilder
-                .image(value.asset)
-                .forceDownload(
-                  assetDocument.originalFilename || `download.${assetDocument.extension}`
-                )
-                .url()}
-              copyUrl={imageUrlBuilder.image(value.asset).url()}
-              readOnly={readOnly}
-              directUploads={directUploads}
-              accept={accept}
-            />
-          </ImageActionsMenu>
-        )}
+        {(assetDocument) => {
+          let copyUrl: string | undefined
+          let downloadUrl: string | undefined
+
+          if (isImageSource(value)) {
+            const filename = assetDocument.originalFilename || `download.${assetDocument.extension}`
+            downloadUrl = imageUrlBuilder.image(assetDocument._id).forceDownload(filename).url()
+            copyUrl = imageUrlBuilder.image(assetDocument._id).url()
+          }
+
+          return (
+            <ImageActionsMenu
+              isMenuOpen={isMenuOpen}
+              onEdit={this.handleOpenDialog}
+              showEdit={showAdvancedEditButton}
+              onMenuOpen={(isOpen) => this.setState({isMenuOpen: isOpen})}
+            >
+              <ActionsMenu
+                onUpload={this.handleSelectFiles}
+                browse={browseMenuItem}
+                onReset={this.handleRemoveButtonClick}
+                downloadUrl={imageUrlBuilder
+                  .image(value.asset)
+                  .forceDownload(
+                    assetDocument.originalFilename || `download.${assetDocument.extension}`
+                  )
+                  .url()}
+                copyUrl={imageUrlBuilder.image(value.asset).url()}
```

Comment:
Since you've moved these out of the return (which is great!) I think we can just replace them with the set values on line 567 and 568 :)

Reply:
Woops, forgot to move it